controllers/openstackcluster_controller.go

```diff
@@ -316,36 +334,21 @@ func deleteBastion(scope *scope.WithLogger, cluster *clusterv1.Cluster, openStac
 				}
 			}
 		}
+	}
 
-		if err = computeService.DeleteInstance(openStackCluster, instanceStatus); err != nil {
+	if bastionServer != nil {
+		if err = r.deleteBastionServer(ctx, openStackCluster, cluster); err != nil {
 			handleUpdateOSCError(openStackCluster, fmt.Errorf("failed to delete bastion: %w", err))
 			return fmt.Errorf("failed to delete bastion: %w", err)
 		}
+		openStackCluster.Status.Bastion = nil
```

Comment:
I need to investigate that. This breaks the api tests: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/2067/pull-cluster-api-provider-openstack-test/1798438671754465280